lua/bufferline/render.lua

```diff
@@ -82,11 +82,13 @@ local function render()
   end
 
   local opts = vim.g.bufferline
-  local icons = vim.g.icons
+
+  -- Ensure that the new setting scheme is adhered to.
+  local icons = type(opts.icons) == 'table' and opts.icons or vim.fn['bufferline#init_icons']()
 
   local click_enabled = vim.fn.has('tablineat') and opts.clickable
-  local has_icons = opts.icons ~= false
-  local has_numbers = opts.icons == 'numbers'
+  local has_icons = icons.devicons == true
```

Comment:
I changed `~= false` to `== true` here, since `devicons` can no longer hold the `'numbers'` string value.